Core/include/Acts/Geometry/NavigationState.hpp

```diff
@@ -0,0 +1,141 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Surfaces/BoundaryCheck.hpp"
+#include "Acts/Utilities/Delegate.hpp"
+#include "Acts/Utilities/Intersection.hpp"
+
+#include <any>
+#include <vector>
+
+/// @note this is foreseen for the 'Geometry' module
+
+namespace Acts {
+
+class Surface;
+
+namespace Experimental {
+
+class Portal;
+class Detector;
+class DetectorVolume;
+
+/// @brief A navigation state struct that is
+/// holding the current navigation information
+/// about volume, surfaces, and portals
+struct NavigationState {
+  /// @brief  A surface candidate and its intersection
+  ///
+  /// candidates can either be surfaces or portals (which contain a surface)
+  struct SurfaceCandidate {
+    /// A candidate intersection, in Surface view
+    ObjectIntersection<Surface> objectIntersection;
+    /// A candidate is either a detector Surface
+    const Surface* surface = nullptr;
+    /// Or a portal
+    const Portal* portal = nullptr;
+    /// The boundary check used for the candidate, boundary checks
+    /// can differ for sensitive surfaces and portals
+    BoundaryCheck bCheck = true;
+  };
+
+  /// Surface candidate vector alias, this allows to use e.g. boost_small vector
+  /// or other stl like containers
+  using SurfaceCandidates = std::vector<SurfaceCandidate>;
+
+  /// The current position
+  Vector3 position = Vector3(0., 0., 0.);
```

Comment:
Hmm just to mention this again: I think for this we already have a stepper interface that exposes this in a defined way without needing to know the details of the stepper-state implementation. This interface is also enforced by a one of these C++17 `static_assert`-concepts, so this should be reliable.

Wouldn't it be an option to also pass stepper and stepper state to the delegate? Since this in principle redundant copy of these information must be done every step?

And e.g. for the `MultiEigenStepper` this is not so cheap, because there we potentially perform a weighted average of multiple components to compute these values, and this would be computed regardless if it is needed...

Reply:
I agree, let's visit this again then.